core/src/main/java/org/apache/iceberg/UpdateSnapshotReferencesOperation.java

```diff
@@ -85,16 +86,29 @@ public UpdateSnapshotReferencesOperation removeTag(String name) {
   }
 
   public UpdateSnapshotReferencesOperation renameBranch(String name, String newName) {
-    Preconditions.checkNotNull(name, "Branch to rename cannot be null");
-    Preconditions.checkNotNull(newName, "New branch name cannot be null");
+    renameRef(name, newName, SnapshotRefType.BRANCH);
+    return this;
+  }
+
+  public UpdateSnapshotReferencesOperation renameTag(String name, String newName) {
+    renameRef(name, newName, SnapshotRefType.TAG);
+    return this;
+  }
+
+  private void renameRef(String name, String newName, SnapshotRefType type) {
```

Comment:
Still don't really know about this helper method. I added it to reduce code duplication for renameBranch/renameTag as they are basically the same (check if the ref exists and it's the expected type. Then verify the target ref doesn't already exist. Remove the old ref and add the new ref), it's just the messaging that's different and we expose the branch/tag specific methods so it's more explicit to the user of the API .

Code duplication is reduced but there's branching logic just for the different names in the message which seems overkill. May be worth the code duplication if the community feels it's more readable that way